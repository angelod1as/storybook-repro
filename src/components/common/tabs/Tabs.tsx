import {
  Root,
  Item,
  ToggleGroupSingleProps as RadixToggleGroupSingleProps,
  ToggleGroupMultipleProps as RadixToggleGroupMultipleProps,
  ToggleGroupItemProps as RadixToggleGroupItemProps,
} from "@radix-ui/react-toggle-group"
import { Children, FC } from "react"

import { create } from "src/helpers/bem"

import styles from "./Tabs.module.scss"

const bem = create(styles, "Tabs")

export type TabsProps =
  | RadixToggleGroupSingleProps
  | RadixToggleGroupMultipleProps
export type TabItemProps = RadixToggleGroupItemProps

export const TabItem: FC<TabItemProps> = ({
  className,
  children,
  ...otherProps
}) => {
  const childrenArray = Children.toArray(children)
  const isStringChild =
    childrenArray.length === 1 && typeof childrenArray[0] === "string"

  return (
    <Item className={bem("item", undefined, className)} {...otherProps}>
      {isStringChild ? (
        <div className={bem("item__label")}>{children}</div>
      ) : (
        children
      )}
    </Item>
  )
}

export const Tabs: FC<TabsProps> = ({ className, ...otherProps }) => {
  return (
    <Root className={bem(undefined, undefined, className)} {...otherProps} />
  )
}
