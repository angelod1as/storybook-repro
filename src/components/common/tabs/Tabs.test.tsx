import { render } from "src/tests"

import { Tabs, TabItem } from "./Tabs"

describe("The toggle group component", () => {
  it("should render single select toggle group", () => {
    const { container } = render(
      <Tabs type="single" defaultValue="Hi">
        <TabItem value="Hello">Hello</TabItem>
        <TabItem value="Hi">Hi</TabItem>
        <TabItem value="Goodbye">Goodbye</TabItem>
      </Tabs>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it("should render multiple select toggle group", () => {
    const { container } = render(
      <Tabs type="multiple" defaultValue={["Hi", "Goodbye"]}>
        <TabItem value="Hello">Hello</TabItem>
        <TabItem value="Hi">Hi</TabItem>
        <TabItem value="Goodbye">Goodbye</TabItem>
      </Tabs>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it("should render toggle group with custom items", () => {
    const { container } = render(
      <Tabs type="single" defaultValue="circle">
        <TabItem value="square">
          <div
            style={{
              width: "10px",
              height: "10px",
              background: "red",
              margin: "10px",
            }}
          />
        </TabItem>
        <TabItem value="circle">
          <div
            style={{
              width: "10px",
              height: "10px",
              background: "blue",
              borderRadius: "50%",
              margin: "10px",
            }}
          />
        </TabItem>
        <TabItem value="triangle">
          <div
            style={{
              width: 0,
              height: 0,
              borderLeft: "5px solid transparent",
              borderRight: "5px solid transparent",
              borderBottom: "10px solid green",
              margin: "10px",
            }}
          />
        </TabItem>
      </Tabs>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it("should render toggle group with no selected value", () => {
    const { container } = render(
      <Tabs type="single">
        <TabItem value="Hello">Hello</TabItem>
        <TabItem value="Hi">Hi</TabItem>
        <TabItem value="Goodbye">Goodbye</TabItem>
      </Tabs>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
