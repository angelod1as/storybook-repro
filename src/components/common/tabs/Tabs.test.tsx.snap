// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The toggle group component should render multiple select toggle group 1`] = `
<div
  class="Tabs"
  dir="ltr"
  role="group"
  style="outline: none;"
  tabindex="0"
>
  <button
    aria-pressed="false"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="off"
    tabindex="-1"
    type="button"
  >
    <div
      class="Tabs__item__label"
    >
      Hello
    </div>
  </button>
  <button
    aria-pressed="true"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="on"
    tabindex="-1"
    type="button"
  >
    <div
      class="Tabs__item__label"
    >
      Hi
    </div>
  </button>
  <button
    aria-pressed="true"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="on"
    tabindex="-1"
    type="button"
  >
    <div
      class="Tabs__item__label"
    >
      Goodbye
    </div>
  </button>
</div>
`;

exports[`The toggle group component should render single select toggle group 1`] = `
<div
  class="Tabs"
  dir="ltr"
  role="group"
  style="outline: none;"
  tabindex="0"
>
  <button
    aria-checked="false"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="off"
    role="radio"
    tabindex="-1"
    type="button"
  >
    <div
      class="Tabs__item__label"
    >
      Hello
    </div>
  </button>
  <button
    aria-checked="true"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="on"
    role="radio"
    tabindex="-1"
    type="button"
  >
    <div
      class="Tabs__item__label"
    >
      Hi
    </div>
  </button>
  <button
    aria-checked="false"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="off"
    role="radio"
    tabindex="-1"
    type="button"
  >
    <div
      class="Tabs__item__label"
    >
      Goodbye
    </div>
  </button>
</div>
`;

exports[`The toggle group component should render toggle group with custom items 1`] = `
<div
  class="Tabs"
  dir="ltr"
  role="group"
  style="outline: none;"
  tabindex="0"
>
  <button
    aria-checked="false"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="off"
    role="radio"
    tabindex="-1"
    type="button"
  >
    <div
      style="width: 10px; height: 10px; background: red; margin: 10px;"
    />
  </button>
  <button
    aria-checked="true"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="on"
    role="radio"
    tabindex="-1"
    type="button"
  >
    <div
      style="width: 10px; height: 10px; background: blue; border-radius: 50%; margin: 10px;"
    />
  </button>
  <button
    aria-checked="false"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="off"
    role="radio"
    tabindex="-1"
    type="button"
  >
    <div
      style="width: 0px; height: 0px; border-left: 5px solid transparent; border-right: 5px solid transparent; border-bottom: 10px solid green; margin: 10px;"
    />
  </button>
</div>
`;

exports[`The toggle group component should render toggle group with no selected value 1`] = `
<div
  class="Tabs"
  dir="ltr"
  role="group"
  style="outline: none;"
  tabindex="0"
>
  <button
    aria-checked="false"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="off"
    role="radio"
    tabindex="-1"
    type="button"
  >
    <div
      class="Tabs__item__label"
    >
      Hello
    </div>
  </button>
  <button
    aria-checked="false"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="off"
    role="radio"
    tabindex="-1"
    type="button"
  >
    <div
      class="Tabs__item__label"
    >
      Hi
    </div>
  </button>
  <button
    aria-checked="false"
    class="Tabs__item"
    data-radix-collection-item=""
    data-state="off"
    role="radio"
    tabindex="-1"
    type="button"
  >
    <div
      class="Tabs__item__label"
    >
      Goodbye
    </div>
  </button>
</div>
`;
